src/metabase/driver/druid.clj

```diff
@@ -70,9 +71,13 @@
                                       :settings (:settings query))))
 
 (defn- process-native [{database-id :database, {query :query} :native}]
-  {:pre [(integer? database-id)]}
-  (let-404 [details (sel :one :field [Database :details] :id database-id)]
-    (do-query details query)))
+  {:pre [(integer? database-id) (string? query)]}
+  (let-404 [details (sel :one :field [Database :details], :id database-id)]
+    (let [query (json/parse-string query keyword)]
+      ;; `annotate` happens automatically as part of the QP middleware for MBQL queries but not for native ones.
+      ;; This behavior was originally so we could preserve column order for raw SQL queries.
+      ;; Since Druid results come back as maps for each row there is no order to preserve so we can go ahead and re-use the MBQL-QP annotation code here.
+      (annotate/annotate query (qp/post-process-native query (do-query details query))))))
```

Comment:
Like my comment above says, certain code pathways (most notably, native SQL) don't use the `annotate` function because they preserve ordering of results instead of the magic column ordering that MBQL queries get. We can't run `annotate` on the results on a native SQL query.